src/JSInterop/Microsoft.JSInterop/test/Infrastructure/ByteArrayJsonConverterTest.cs

```diff
@@ -0,0 +1,157 @@
+            Assert.Equal("Required property __byte[] not found.", ex.Message);
+        }
+
+        [Fact]
+        public void Read_Throws_IfJsonIsIncomplete()
+        {
+            // Arrange
+            JSRuntime.ByteArraysToBeRevived.Append(new byte[] { 1, 2 });
+
+            var json = $"{{\"__byte[]\":0";
+
+            // Act & Assert
+            var ex = Record.Exception(() => JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions));
+            Assert.IsAssignableFrom<JsonException>(ex);
+        }
+
+        [Fact]
+        public void Read_ReadsJson()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json = $"{{\"__byte[]\":0}}";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Equal(byteArray, deserialized);
+        }
+
+        [Fact]
+        public void Read_IfByteArraysIdAppearsMultipleTimesUseLastProperty()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json = $"{{\"__byte[]\":9120,\"__byte[]\":0}}";
+
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Act & Assert
+            Assert.Equal(byteArray, deserialized);
+        }
+
+        [Fact]
+        public void Read_ReturnsTheCorrectInstance()
+        {
+            // Arrange
+            // Track a few arrays and verify that the deserialized value returns the correct value.
+            var byteArray1 = new byte[] { 1, 5, 7 };
+            var byteArray2 = new byte[] { 2, 6, 8 };
+            var byteArray3 = new byte[] { 2, 6, 8 };
+
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray1);
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray2);
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray3);
+
+            var json = $"[{{\"__byte[]\":2}},{{\"__byte[]\":1}}]";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[][]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Same(byteArray3, deserialized[0]);
+            Assert.Same(byteArray2, deserialized[1]);
+        }
+
+        [Fact]
+        public void Read_ReadsJson_WithFormatting()
+        {
+            // Arrange
+            var byteArray = new byte[] { 1, 5, 7 };
+            JSRuntime.ByteArraysToBeRevived.Append(byteArray);
+
+            var json =
+@$"{{
+            ""__byte[]"": 0
+        }}";
+
+            // Act
+            var deserialized = JsonSerializer.Deserialize<byte[]>(json, JsonSerializerOptions)!;
+
+            // Assert
+            Assert.Equal(byteArray, deserialized);
+        }
```

Comment:
It's good that you thought of this. That said, this is testing system.text.json which we shouldn't just do here. It's fine if we remove it.